src/channel/set.rs

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MIT
+
+use futures::TryStream;
+use netlink_packet_generic::GenlMessage;
+
+use crate::{ethtool_execute, EthtoolAttr, EthtoolError, EthtoolHandle, EthtoolMessage, EthtoolChannelAttr};
+
+pub struct EthtoolChannelSetRequest {
+    handle: EthtoolHandle,
+    message: EthtoolMessage,
+}
+
+impl EthtoolChannelSetRequest {
+    pub(crate) fn new(handle: EthtoolHandle, iface_name: &str) -> Self {
+        EthtoolChannelSetRequest {
+            handle,
+            message: EthtoolMessage::new_channel_set(iface_name),
+        }
+    }
+
+    pub fn rx_count(mut self, count: u32) -> Self {
```

Comment:
Instead of every function pushing a NLA, could you try builder way like https://github.com/rust-netlink/rtnetlink/pull/57